### ISSUE TITLE
Lint-as for/doseq doesn't support `:let` bindings

### ISSUE THREAD

**Issue by Deraen** [borkdude/clj-kondo#434]
**version**

v2019.08.21-alpha

**platform**

Linux

**problem**

Using `:lint-as` to declare a macro to work like `for` or `doseq` doesn't still add support for using `:let` bindings inside the macro.

**repro**

```
clj-kondo --config '{:lint-as {plumbing.core/for-map clojure.core/doseq}}' --lint - <<EOF
(require 'plumbing.core)
(plumbing.core/for-map [[k v] {} :let [[v & _rst] v]] k v)
(for [[k v] {} :let [[v & _rst] v]] [k v])
EOF
```

**expected behavior**

`for-map` works like `for`, no warnings.

**actual behavior**

Warning about unresolved symbols for `&` and `_rst` in `for-map` `:let` binding.

```
<stdin>:2:43: info: unresolved symbol &
<stdin>:2:45: info: unresolved symbol _rst
linting took 16ms, errors: 0, warnings: 0
```

**Comment by borkdude** [comment]
Hi @Deraen, this is fixed in https://github.com/borkdude/clj-kondo/commit/8a2df7b75650672aff21334d3a42a4f911802443 so you can test it either using that commit or using this Linux binary produced on CircleCI:

https://5483-176829714-gh.circle-artifacts.com/0/release/clj-kondo-2019.08.22-alpha-SNAPSHOT-linux-amd64.zip

**Comment by Deraen** [comment]
This case works now.

Though, I think cause of the fix I now get another false positive from a quite different case:

```
❯ clj-kondo --config '{:lint-as {clojure.java.jdbc/with-db-transaction clojure.core/let}}' --lint - <<EOF
(require 'clojure.java.jdbc)
(let [a 1 db 2]
  (clojure.java.jdbc/with-db-transaction [db db] db))
EOF
<stdin>:3:3: warning: redundant let
linting took 11ms, errors: 0, warnings: 1
```

Now let-like binding from with-db-transaction incorrectly reported as reduntant.

**Comment by borkdude** [comment]
Thanks, I'll look into it.

**Comment by borkdude** [comment]
@Deraen Thanks for testing. I've made another fix:

https://5491-176829714-gh.circle-artifacts.com/0/release/clj-kondo-2019.08.22-alpha-SNAPSHOT-linux-amd64.zip

Commit SHA: 2e452ffd77d77bcb1eda58344fc05586b587b45c

Can you test once more?

**Comment by Deraen** [comment]
Works now, thanks!

Interesting to look at the fixes for these problems!
